operator/webhook/validate.go

```diff
@@ -0,0 +1,28 @@
+package webhook
+
+import (
+	"fmt"
+
+	"github.com/vshn/provider-exoscale/operator/mapper"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func ValidateRawExtension(raw runtime.RawExtension) error {
+	m, err := mapper.ToMap(raw)
+	if err != nil {
+		return fmt.Errorf("mapper.ToMap(%q): %w", raw, err)
+	}
+	for k, v := range m {
+		switch v.(type) {
+		case string:
+		case int64:
+		case float64:
+		case bool:
+			continue
+		default:
+			return fmt.Errorf("validate: value of key %q is not a supported type (only strings, boolean and numbers): %v", k, v)
+		}
```

Comment:
```
unable to create instance: Post \"https://api-ch-dk-2.exoscale.com/v2/dbaas-redis/redis-local-dev\": invalid request: Incorrect settings: Invalid input for redis_maxmemory_policy: {'test': 'ing'} is not of type 'string', 'null'"}
```

when submitting:
```
    redisSettings:
      foo:
        - bar
        - baz
      maxmemory_policy: { test: ing }
```

when removing the `maxmemory_policy` that goes through without issues and the setting is just not applied. 
I believe we should leave it at that for the moment because it seems sensible to me and revisit if we need later more. A full coverage of valid values is probably hard to do.

Reply:
fine by me